specs/eid-hermes-commands-format.md

```diff
@@ -0,0 +1,227 @@
+|--------|----------------------------------------|
+| 0x00   | Success                                |
+| 0x01   | All slots of requested type are in use |
+
+**Table 7: Request Slots status codes**
+
+### Release Slots (Opcode 0x01)
+
+This command can be used to release a program or data slot on the device.
+
+It is invalid to release a slot that is not allocated.
+
+| Bytes | Description                           |
+|-------|---------------------------------------|
+| 08    | Slot type (0x00: program, 0x01: data) |
+| 09    | Slot ID                               |
+
+**Table 8: Release Slots Command Request**
+
+| Bytes | Description                |
+|-------|----------------------------|
+
+(no command-specific bytes used)
+
+**Table 9: Release Slots Command Response**
+
+This command may return the following status:
+
+| Status | Description                                                |
+|--------|------------------------------------------------------------|
+| 0x00   | Success                                                    |
+| 0x02   | Requested program slot does not exist or is not allocated  |
+| 0x03   | Requested data slot does not exist or is not allocated     |
+
+**Table 10: Release Slots status codes**
+
+### Write to Slot (Opcode 0x10)
+
+This command can be used to write to a program or data slot. The slot must have
+been previously requested with opcode 0x00 or this will fail.
+
+This command does not guarantee that all bytes will be transferred. The number
+of actual bytes transferred is reported on the Command Response.
+
+| Bytes | Description                               |
+|-------|-------------------------------------------|
+| 08    | Slot type (0x00: program, 0x01: data)     |
+| 09    | Slot ID                                   |
+| 11:10 | Reserved                                  |
+| 19:12 | Source Addr: the host address to transfer |
+| 23:20 | Data Length                               |
+
+**Table 11: Write to Slot Command Request**
+
+| Bytes | Description                                             |
+|-------|---------------------------------------------------------|
+| 11:08 | Number of bytes written. Only valid when Status is 0x00 |
+
+**Table 12: Write to Slot Command Response**
+
+| Status | Description                                                    |
+|--------|----------------------------------------------------------------|
+| 0x00   | Success. Does not guarantee that all data has been transferred |
+| 0x02   | Requested program slot does not exist or is not allocated      |
+| 0x03   | Requested data slot does not exist or is not allocated         |
+| 0x04   | Invalid source address                                         |
+
+**Table 13: Write to Slot status codes**
+
+### Read Data from Slot (Opcode 0x11)
+
+This command can be used to read data from a slot. The slot must have been
+previously requested with opcode 0x00 or this will fail.
+
+Only data slots can be read.
```

Comment:
Why only data slots? And again we can do this via XDMA and we want to keep this interface as simple as possible so we should avoid re-implementing things we can do via the XDMA?

Reply:
I restricted to data slots only because I can't see a valid use-case for reading a program slot

Reply:
@iomartin I think we want to be able to read back what is in a program slot for verification and debug etc.